src/nvim/option.c

```diff
@@ -180,14 +180,37 @@ static void set_init_default_shell(void)
   // Find default value for 'shell' option.
   // Don't use it if it is empty.
   const char *shell = os_getenv("SHELL");
+
   if (shell != NULL) {
+  char * fixed_shell = NULL;
+
+  // In case of using backslashes in SHELL path switch them to forward slashes
+  // so given shell can be run.
+#ifdef BACKSLASH_IN_FILENAME
+  if(vim_strchr(shell, '\\') != NULL) {
```

Comment:
is this the inverse of slash_adjust? https://github.com/neovim/neovim/blob/5d4d5b5d54dc2e46733d25bfa2ee80a35db76b39/src/nvim/path.c#L1423

inlining a random special case in `set_init_default_shell` doesn't seem very maintainable.

also needs a test.

Reply:
Thank you for your comment, sooo how would you implement it differently?

Reply:
Does changing `ex_terminal()` (e.g. <https://github.com/neovim/neovim/issues/16646#issuecomment-993599308>) not work?